### ISSUE TITLE
server version errors for gke 1-2 ingress tests

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#24837]
gce-1.2 is fine: http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-ingress-release-1.2/
but gke-1.2 is failing since friday with:
```
error: server version (version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.4-beta.0-dirty", GitCommit:"b14ae23f32077d8911f88bc47ab15842fb93b339", GitTreeState:"dirty"}) differs from client version (version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.4-beta.0", GitCommit:"b14ae23f32077d8911f88bc47ab15842fb93b339", GitTreeState:"clean"})!
15:04:56 2016/04/22 15:04:56 e2e.go:200: Error running get status: exit status 1
15:04:56 2016/04/22 15:04:56 e2e.go:196: Step 'get status' finished in 206.829821ms
15:04:56 2016/04/22 15:04:56 e2e.go:181: Testing requested, but e2e cluster not up!
```
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gke-ingress-release-1.2/1564/console
Anything changed recently?
@kubernetes/goog-testing

**Comment by fejta** [comment]
I believe @Q-Lee was talking about this today at lunch?

**Comment by ixdy** [comment]
Something looks weird with the release-1.2 build.

I downloaded and extracted `gs://kubernetes-release/ci/v1.2.4-beta.0/kubernetes-client-linux-amd64.tar.gz` and `gs://kubernetes-release/release/v1.2.4-beta.0/kubernetes-server-linux-amd64.tar.gz`.

```console
$ ./kubernetes/client/bin/kubectl version
Client Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.4-beta.0", GitCommit:"b14ae23f32077d8911f88bc47ab15842fb93b339", GitTreeState:"clean"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
$ ./kubernetes/server/bin/kubectl version
Client Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.4-beta.0-dirty", GitCommit:"b14ae23f32077d8911f88bc47ab15842fb93b339", GitTreeState:"dirty"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

I'd expect these to be the same, but one is reporting as "dirty".

**Comment by ixdy** [comment]
Wait, I think I see what's wrong. I accidentally downloaded one tarball from `ci/` and one from `release/`, which is probably what the gke job is doing, too.

The tarballs under `ci/` look fine. The `v1.2.4-beta.0` tarballs under `release/` have the `-dirty` bit.

I'm guessing the GCE release-1.2 jobs are using the `ci/` tarballs for everything, so they're working fine. The GKE release-1.2 jobs are probably using the `ci/` tarball for client binaries, but the `release/` tarball for the server binaries. Since these don't match, it's failing.

It looks like `release/v1.2.3` is clean; it's only `release/v.1.2.4-beta.0` which is dirty.

@david-mcmahon

**Comment by david-mcmahon** [comment]
I will look at this tonight.

**Comment by david-mcmahon** [comment]
Let's try that again.  I'll get a chance today to look.

**Comment by david-mcmahon** [comment]
I assume these beta.0 releases are only used by the various internal testing frameworks as it's identical to the official version that accompanies it.  The issue is there was an extra _output* directory when this was built, nothing more, causing 'git status --porcelain' to complain and append -dirty.  Unless anyone objects, I'll just rebuild 1.2.4-beta.0 and upload it.

**Comment by david-mcmahon** [comment]
Uploaded corrected 1.2.4-beta.0 binaries.
